crates/world/src/conversions.rs

```diff
@@ -100,6 +220,106 @@ mod rdbms_types {
+                    spin::postgres::postgres::ParameterValue::Int32(i)
+                }
+                v1::rdbms_types::ParameterValue::Int64(i) => {
+                    spin::postgres::postgres::ParameterValue::Int64(i)
+                }
+                v1::rdbms_types::ParameterValue::Uint8(_)
+                | v1::rdbms_types::ParameterValue::Uint16(_)
+                | v1::rdbms_types::ParameterValue::Uint32(_)
+                | v1::rdbms_types::ParameterValue::Uint64(_) => {
+                    return Err(v1::postgres::PgError::ValueConversionFailed(
+                        "Postgres does not support unsigned integers".to_owned(),
+                    ));
+                }
+                v1::rdbms_types::ParameterValue::Floating32(r) => {
+                    spin::postgres::postgres::ParameterValue::Floating32(r)
+                }
+                v1::rdbms_types::ParameterValue::Floating64(r) => {
+                    spin::postgres::postgres::ParameterValue::Floating64(r)
+                }
+                v1::rdbms_types::ParameterValue::Str(s) => {
+                    spin::postgres::postgres::ParameterValue::Str(s)
+                }
+                v1::rdbms_types::ParameterValue::Binary(b) => {
+                    spin::postgres::postgres::ParameterValue::Binary(b)
+                }
+                v1::rdbms_types::ParameterValue::DbNull => {
+                    spin::postgres::postgres::ParameterValue::DbNull
+                }
+            };
+            Ok(converted)
+        }
+    }
+
+    impl TryFrom<v2::rdbms_types::ParameterValue> for spin::postgres::postgres::ParameterValue {
+        type Error = v2::rdbms_types::Error;
+
+        fn try_from(
+            value: v2::rdbms_types::ParameterValue,
+        ) -> Result<spin::postgres::postgres::ParameterValue, Self::Error> {
+            let converted = match value {
+                v2::rdbms_types::ParameterValue::Boolean(b) => {
+                    spin::postgres::postgres::ParameterValue::Boolean(b)
+                }
+                v2::rdbms_types::ParameterValue::Int8(i) => {
+                    spin::postgres::postgres::ParameterValue::Int8(i)
+                }
+                v2::rdbms_types::ParameterValue::Int16(i) => {
+                    spin::postgres::postgres::ParameterValue::Int16(i)
+                }
+                v2::rdbms_types::ParameterValue::Int32(i) => {
+                    spin::postgres::postgres::ParameterValue::Int32(i)
+                }
+                v2::rdbms_types::ParameterValue::Int64(i) => {
+                    spin::postgres::postgres::ParameterValue::Int64(i)
+                }
+                v2::rdbms_types::ParameterValue::Uint8(_)
+                | v2::rdbms_types::ParameterValue::Uint16(_)
+                | v2::rdbms_types::ParameterValue::Uint32(_)
+                | v2::rdbms_types::ParameterValue::Uint64(_) => {
+                    return Err(v2::rdbms_types::Error::ValueConversionFailed(
```

Comment:
Yes, v2 apps also failed, but later, during the "convert `rdbms-types` type to Postgres type" stage (https://github.com/fermyon/spin/blob/3eaba5fdae634ad000b8268b7687ba24d7d752b4/crates/factor-outbound-pg/src/client.rs#L123).  It's possible there's a change to the error text or the way it's reported, but since this was a "type error at runtime," I'm reasonably confident that nobody was relying on the specifics of unsigned type errors - if they had an unsigned type in Postgres code then they ripped the type out rather than handling the error.

If you or other folks feel that's an unreasonable judgment, I can look at other ways to tackle this!  But I'm definitely keen to keep unsigned types out of the v3 interface.

Reply:
Failing earlier sounds good to me and i agree that folks relying on a specific error does not seem to outweigh the benefit of removing unsigned types